pkgs/top-level/python-packages.nix

```diff
@@ -4196,6 +4196,28 @@ in modules // {
       description = "pytest plugin with mechanisms for caching across test runs";
     };
   };
+  
+  pytestdjango = buildPythonPackage rec {
+    name = "pytest-django-${version}";
+    version = "2.9.1";
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/p/pytest-django/${name}.tar.gz";
+      sha256 = "1mmc7zsz3dlhs6sx4sppkj1vgshabi362r1a8b8wpj1qfximpqcb";
+    };
+    
+    # doing this to allow depending packages to find
+    # pytest's binaries
+    pytest = self.pytest;
```

Comment:
Nix builds up a graph of all dependencies. You've stated that `pytest` is a dependency of `pytestdjango`.
If another package now depends on `pytestdjango`, it can follow the graph and get that specific `pytest` version.

With Python, in order for packages to be visible, they need to be added to PYTHONPATH. If you now create an environment with multiple versions of `pytest`, e.g. mixing the 2.7 and 3.5 versions (note, they are different nodes in the Nix graph, since `pkgs.python27Packages != pkgs.python35Packages`), then they are both added to PYTHONPATH. This is a problem. Which version should Python now import? This is actually a general problem with Python and the reason why `virtualenv` is used.

You gave the `buildPythonPackage`, Python definitions, the keyword argument `pytest = self.pytest`, but this parameter doesn't exist and is ignored.